server/src/main/java/org/elasticsearch/transport/RemoteConnectionManager.java

```diff
@@ -95,13 +101,7 @@ public void openConnection(DiscoveryNode node, @Nullable ConnectionProfile profi
             node,
             profile,
             listener.delegateFailureAndWrap(
-                (l, connection) -> l.onResponse(
-                    new InternalRemoteConnection(
-                        connection,
-                        clusterAlias,
-                        profile != null ? profile.getTransportProfile() : getConnectionProfile().getTransportProfile()
-                    )
-                )
+                (l, connection) -> l.onResponse(wrapConnectionWithRemoteClusterInfo(connection, clusterAlias, credentialsManager))
```

Comment:
I've replaced the `new InternalRemoteConnection(...)` with `wrapConnectionWithRemoteClusterInfo` to lock down how an internal remote connection can be instantiated (this is a small refactor unrelated to the functional change in this PR). 

The functional change here is passing `credentialsManager` to the wrap call instead of a fixed boolean flag. I got into more detail on this change in my comment on the `wrapConnectionWithRemoteClusterInfo` method below.